File: bukkit/src/main/java/me/kvalbrus/multibans/bukkit/implementations/BukkitConsole.java
package me.kvalbrus.multibans.bukkit.implementations;

import me.kvalbrus.multibans.api.Console;
import org.bukkit.command.ConsoleCommandSender;
import org.jetbrains.annotations.NotNull;

public class BukkitConsole extends BukkitCommandSender implements Console {

    private final ConsoleCommandSender sender;

    public BukkitConsole(ConsoleCommandSender sender) {
        super(sender);
        this.sender = sender;
    }



    @NotNull
    @Override
    public String getName() {
        return this.sender.getName();
    }

    @Override
    public boolean hasPermission(String permission) {
        return this.sender.hasPermission(permission);
    }
}

File: bukkit/src/main/java/me/kvalbrus/multibans/bukkit/implementations/BukkitConsole.kt
package me.kvalbrus.multibans.bukkit.implementations

import me.kvalbrus.multibans.api.Console
import me.kvalbrus.multibans.common.implementations.MultiConsole
import org.bukkit.command.ConsoleCommandSender
import java.util.*

class BukkitConsole : MultiConsole, Console {

    private val sender: ConsoleCommandSender

    constructor(sender: ConsoleCommandSender) {
        this.sender = sender
    }

    override val uniqueId: UUID? = null

    override fun sendMessage(message: String) = sender.sendMessage(message)
}